apps/web/server/routers/viewer.tsx

```diff
@@ -466,9 +466,9 @@ const loggedInViewerRouter = createProtectedRouter()
       }
 
       let nextCursor: typeof skip | null = skip;
-
-      if (bookingsFetched > take) {
-        bookings.shift();
+      if (input.cursor === undefined && bookings.length + groupedRecurringBookings.length < take) {
```

Comment:
With these we handle 3 scenarios

- First fetch, when cursor its undefined and results length are lower than page size (take prop)
- Normal pagination
- End of page